File: js/bubbleSort.js
function bubbleSort(arr) {
  do {
    sort = false;
    for (var i = 0; i < arr.length-1; i+= 1) {
      if (arr[i] > arr[i+1]) {
        sort = true;
        arr[i] =  arr[i] ^ arr[i+1];
        arr[i+1] = arr[i] ^ arr[i+1];
        arr[i] =  arr[i] ^ arr[i+1];
      }
    }
  } while (sort);
  return arr;
}

module.exports = bubbleSort;

File: python/bubbleSort.py
def bubbleSort(lst):
    sort = False
    while not sort:
        sort = True
        for i in range(len(lst) - 1):
            if lst[i] > lst[i + 1]:
                sort = False
                lst[i] = lst[i] ^ lst[i + 1]
                lst[i + 1] = lst[i] ^ lst[i + 1]
                lst[i] = lst[i] ^ lst[i + 1]
    return lst
